packages/app/src/components/authorize/AuthorizeToken.tsx

```diff
@@ -0,0 +1,50 @@
+import { FC, MouseEventHandler } from "react";
+import { Alert, Tooltip, OverlayTrigger } from "react-bootstrap";
+import { constants } from "ethers";
+import { addresses } from "@tender/contracts";
+import { BigNumber } from "@ethersproject/bignumber";
+
+type Props = {
+  protocolName: string;
+  tokenSymbol: string;
+  isTokenAuthorized: boolean;
+  approveToken: (contract: string, amount: BigNumber) => Promise<void>;
```

Comment:
This doesn't have to be a prop instead you need to pass the arguments as prop as well as the address of the token to be approved and you create the contract call in this component...

Reply:
Ideally you want to keep your leaf components dummy (view components), and have the logic somewhat higher. We can wrap this into another component which does the contract call, but it feels over-engineering at this point.

Reply:
That goes against most engineering rules. If logic can be isolated to a single function, class, component, whatever it should be done so. Since the only variables here are the token address and the arguments for `approve()` it should be isolated within the component.

Reply:
I'm okay with your approach as well, but the above is just not true for react, because components are often split to presentational and container components, where the former does not contain application logic, just view. But as I said I'm okay implementing it either way.

Reply:
RIght , I understand what you're saying. But then the logic should still be isolated where possible as well so wouldn't it end up being something like this still ?  

```
approve/
   index.tsx (logic)
   view.tsx (view)
```